### PR TITLE
VEP 110 & cache process updates

### PR DESCRIPTION
**Pull request by MattWellie**
Most of the contributing line count is moving the current contents of `vep` into `vep/previous_process`. Actual changes:

1. `vep/README.md`, new file. Documents a cheap asynchronous replacement process for acquiring VEP cache files
2. `vep/prod_sync_script.sh`, new file. Bash script to gcloud rsync cache files to `main` following successful testing. Simple checks on presence at source & absence at destination to prevent data loss.
3. `references.py`, modified. Adds docstrings to please the pre-commit hook, and adds a location (without pulling data from new source) for the VEP 110 cache files. Aim is that the above script would be used to rsync data over, and this would only populate the same entity in the config file.

This README is a documentation of the process I've followed, based on aiming to reduce costs without sacrificing on content. Please check if this process seems ridiculous, and whether it could be easily replicated by another user. 

e.g. Globus transfers require creating an account, but the data is public. It also requires installing an app, but we have permission to do so on local laptops. Download/upload is slow, but with Globus and rsync it's interruptible without needing to repeat from scratch. 

This does not address the actual testing of the VEP installation/data. That remains the concern of a pipeline testing process &/or the use of the VEP components in isolation (e.g. [here](https://github.com/populationgenomics/analysis-runner/tree/main/scripts/vep_backend), code which I am aiming to migrate to [here](https://github.com/populationgenomics/production-pipelines/tree/main/scripts/vep_backend))

### CODE REVIEW

File: vep/README.md [hunk 1]
```diff
@@ -1,77 +1,110 @@
...
-The Query's `hl.vep` function works only with the Spark backend on a Dataproc cluster. To make it work with a custom VEP version, the following extra steps are required.
-
-1. The dataproc initialisation script in this repository [dataproc-init.sh](dataproc-init.sh) is copied from the [Hail codebase](https://github.com/hail-is/hail/blob/cc0a051740f4de08408e6a2094ffcb1c3158ee9c/hail/python/hailtop/hailctl/dataproc/resources/vep-GRCh38.sh) and adjusted to pull the image from CPG artefact registry, and parameterised by `VEP_VERSION`.
-
-Script is parametrised with `__VEP_VERSION__`, so pass it through `sed` when copying to the bucket:
-
-```shell
-cat dataproc-init.sh | sed "s/__VEP_VERSION__/${VEP_VERSION}/g" | gsutil cp - gs://cpg-common-main/references/vep/${VEP_VERSION}/dataproc/init.sh
+```commandline
+tar xzf homo_sapiens_vep_110_GRCh38.tar.gz
```
Comment (illusional) [inline vep/README.md:50]: How big does this file roughly become, should we recommend a user has at least `X GB` space available?

File: vep/README.md [hunk 2]
```diff
@@ -1,77 +1,110 @@
...
+then initiates the rsync between the two locations. This should be run using analysis-runner, at the
+`full` access level:
+
+```bash
+analysis-runner \\
+    --dataset seqr \\
+    --access-level full \\
+    --description "VEP: Sync 110 to production" \\
+    --output-dir vep_110 \\
+    bash scripts/prod_sync_script.sh 110
```
Comment (illusional) [inline vep/README.md:109]:
```suggestion
analysis-runner \
    --dataset common \
    --access-level full \
    --description "VEP: Sync 110 to production" \
    --output-dir vep_110 \
    bash scripts/prod_sync_script.sh 110
```

**Review by illusional — COMMENTED** [review]
I don't have much context here, but this looks pretty good to me - left a couple of small comments.

File: vep/README.md (hunk 1 above)
Comment (MattWellie) [reply vep/README.md:54]: Added a bit more detail here - it's ~20GB and unpacks to a second 20GB directory, with a little bit of mischief during the expansion.

**Review by violetbrina — APPROVED** [review]
Yeah again not much context on this, but I do like the updated readme.

**Pull request merged**